app/models/repo.rb

```diff
@@ -170,6 +170,11 @@ def self.without_user_subscriptions(user_id)
     self.where.not(id: user_subscribed_repo_ids)
   end
 
+  def self.in_user_lang_preferences(user)
+    return all unless user.has_favorite_languages?
+    self.where('language IN (?)', user.favorite_languages)
```

Comment:
Shouldn't need to go to raw SQL for this,

```ruby
self.where(language: user.favorite_languages)
```
should produce the same result and keep this in full ActiveRecord.